# 01c — Data Inspection (EyePACS download + OLIVES zip exploration)

## Purpose

Pre-preprocessing data inspection covering two distinct tasks:

1. **EyePACS acquisition** — download the EyePACS dataset from HuggingFace and persist it to Google Drive
2. **OLIVES structural exploration** — inspect the nested OLIVES zip archives (olive.zip → OLIVES.zip → trial-level zips) and the label files, without extracting any image data

These tasks are exploratory and non-destructive (except for the EyePACS download and the OLIVES labels zip extraction, which are intentional one-time operations).

## Inputs

- `olive.zip` (33.92 GB) on `MyDrive/` — wrapper containing the nested OLIVES archives
- `OLIVES_Dataset_Labels.zip` (~10 MB) on `MyDrive/` — label files (CSVs, Excel)
- GitHub token in Colab Secrets as `GITHUB_TOKEN`
- Internet access (for HuggingFace EyePACS download)

## Outputs

- EyePACS HuggingFace arrow files saved to `MyDrive/` (one-time download)
- `MyDrive/olives_labels/` — extracted OLIVES label files (CSVs, Excel)
- Console output describing zip structure, file counts, modality breakdown

## Runtime requirements

- CPU runtime sufficient (no GPU needed for inspection)
- HuggingFace download: 10-20 minutes depending on connection
- Zip inspection cells: 2-10 minutes each (reading from a 34 GB file on Drive is slow)


## Session setup

Clone the repo, mount Drive, configure Python path.

In [ ]:
# ── STANDARD SESSION SETUP ── Run this first every session ──
# PURPOSE: Mount Drive, clone or pull the latest repo from GitHub,
# move into the repo, add it to the Python path, and run the
# project setup helper that creates required output directories.
# RUN: First cell of every Colab session.

import os
import sys
from google.colab import drive, userdata

# 1. Mount Drive (idempotent — safe to re-run)
drive.mount('/content/drive')
print("Drive mounted")

# 2. Move to content root
os.chdir('/content')

# 3. Clone or pull repo (token read from Colab Secrets — never hardcoded)
REPO_DIR = '/content/dr-dissertation'
TOKEN = userdata.get('GITHUB_TOKEN')

if not os.path.exists(REPO_DIR):
    !git clone https://savita10:{TOKEN}@github.com/savita10/dr-dissertation.git {REPO_DIR}
    print("Repo cloned")
else:
    !git -C {REPO_DIR} pull
    print("Repo updated")

# 4. Move into repo and add to Python path
os.chdir(REPO_DIR)
sys.path.insert(0, REPO_DIR)
print(f"Working directory: {os.getcwd()}")

# 5. Run project setup helper — creates expected output directories
from src.utils.colab_setup import setup_colab
paths = setup_colab()
print("Setup complete")


Mounted at /content/drive
Drive mounted
Cloning into 'dr-dissertation'...
remote: Enumerating objects: 29, done.
remote: Counting objects: 100% (29/29), done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 29 (delta 6), reused 27 (delta 4), pack-reused 0 (from 0)
Receiving objects: 100% (29/29), 8.11 KiB | 8.11 MiB/s, done.
Resolving deltas: 100% (6/6), done.
Repo cloned
Working directory: /content/dr-dissertation
Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
GPU: Tesla T4 (14.6 GB)
RAM: 12.7 GB total
Disk (/content): 192.7 GB free / 235.7 GB total
Setup complete


In [3]:
# Run inspection in Colab
# Cell 1
%cd /content/dr-dissertation
!git pull

/content
Already up to date.


In [4]:
# Run inspection in Colab
# Cell 2
from src.utils.colab_setup import setup_colab
paths = setup_colab()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
GPU: Tesla T4 (14.6 GB)
RAM: 12.7 GB total
Disk (/content): 192.7 GB free / 235.7 GB total


## Run zip inspection script

Delegates to `src/utils/inspect_zips.py` which performs non-destructive inspection of all source archives.

In [5]:
# Run inspection in Colab
# Cell 3
!python src/utils/inspect_zips.py

ZIP: EYEPACS.zip
Size: 6.03 GB
Total files inside: 35108
Top-level entries:
  eyepacs_preprocess/
First 20 filenames:
  eyepacs_preprocess/eyepacs_preprocess/12769_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/38743_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/32704_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/23497_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/23851_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/29058_left.jpeg
  eyepacs_preprocess/eyepacs_preprocess/18252_left.jpeg
  eyepacs_preprocess/eyepacs_preprocess/37666_left.jpeg
  eyepacs_preprocess/eyepacs_preprocess/42011_left.jpeg
  eyepacs_preprocess/eyepacs_preprocess/40153_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/14835_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/825_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/25851_right.jpeg
  eyepacs_preprocess/eyepacs_preprocess/15544_left.jpeg
  eyepacs_preprocess/eyepacs_preprocess/43040_right.jpeg
  eyepacs_preprocess/eyepacs_prepr

## Download EyePACS from HuggingFace

The EyePACS dataset is hosted on HuggingFace as `bumbledeep/eyepacs` (MIT licence). Download once, then save to Drive so we don't have to re-download in future sessions.

In [6]:
# Install HuggingFace datasets library
!pip install datasets -q

from datasets import load_dataset

# Load EyePACS dataset directly from HuggingFace
dataset = load_dataset("bumbledeep/eyepacs", split="train")

print(f"Total images: {len(dataset)}")
print(f"Columns: {dataset.column_names}")
print(f"Sample row: {dataset[0]['label_code']}, {dataset[0]['label']}")

/usr/local/lib/python3.12/dist-packages/huggingface_hub/utils/_auth.py:93: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


README.md: 0.00B [00:00, ?B/s]

data/train-00000-of-00014.parquet:   0%|          | 0.00/465M [00:00<?, ?B/s]

data/train-00001-of-00014.parquet:   0%|          | 0.00/467M [00:00<?, ?B/s]

data/train-00002-of-00014.parquet:   0%|          | 0.00/466M [00:00<?, ?B/s]

data/train-00003-of-00014.parquet:   0%|          | 0.00/465M [00:00<?, ?B/s]

data/train-00004-of-00014.parquet:   0%|          | 0.00/470M [00:00<?, ?B/s]

data/train-00005-of-00014.parquet:   0%|          | 0.00/467M [00:00<?, ?B/s]

data/train-00006-of-00014.parquet:   0%|          | 0.00/467M [00:00<?, ?B/s]

data/train-00007-of-00014.parquet:   0%|          | 0.00/467M [00:00<?, ?B/s]

data/train-00008-of-00014.parquet:   0%|          | 0.00/467M [00:00<?, ?B/s]

data/train-00009-of-00014.parquet:   0%|          | 0.00/465M [00:00<?, ?B/s]

data/train-00010-of-00014.parquet:   0%|          | 0.00/469M [00:00<?, ?B/s]

data/train-00011-of-00014.parquet:   0%|          | 0.00/464M [00:00<?, ?B/s]

data/train-00012-of-00014.parquet:   0%|          | 0.00/466M [00:00<?, ?B/s]

data/train-00013-of-00014.parquet:   0%|          | 0.00/469M [00:00<?, ?B/s]

Generating train split:   0%|          | 0/35108 [00:00<?, ? examples/s]

Total images: 35108
Columns: ['image', 'label_code', 'label']
Sample row: 0, no_diabetic_retinopathy


## Save EyePACS to Drive

Persists the HuggingFace arrow files to MyDrive root. The preprocessing pipeline reads from this saved copy.

In [7]:
# Save to Drive so you never re-download again
dataset.save_to_disk(
    "/content/drive/MyDrive/"
)
print(f"Saved {len(dataset)} rows to Drive")
print("Location: /content/drive/MyDrive/")

Saving the dataset (0/14 shards):   0%|          | 0/35108 [00:00<?, ? examples/s]

Saved 35108 rows to Drive
Location: /content/drive/MyDrive/


## Basic inspection of OLIVES source zips

Lists file counts, top-level entries, and file extensions for `olive.zip` (wrapper) and `OLIVES_Dataset_Labels.zip`. Non-destructive — only reads zip indices, doesn't extract anything.

In [8]:
import zipfile
import os

zips = [
    "/content/drive/MyDrive/olive.zip",
    "/content/drive/MyDrive/OLIVES_Dataset_Labels.zip"
]

for zip_path in zips:
    print(f"\n{'='*60}")
    print(f"ZIP: {os.path.basename(zip_path)}")
    print(f"Size: {os.path.getsize(zip_path)/1e9:.2f} GB")
    with zipfile.ZipFile(zip_path, 'r') as z:
        files = z.namelist()
        print(f"Total files: {len(files)}")
        top_level = set(f.split('/')[0] for f in files)
        print(f"Top-level entries: {top_level}")
        print(f"First 20 files:")
        for f in files[:20]:
            print(f"  {f}")
        from collections import Counter
        exts = Counter(os.path.splitext(f)[1].lower()
                      for f in files if '.' in f)
        print(f"Extensions: {dict(exts)}")


ZIP: olive.zip
Size: 33.92 GB
Total files: 2
Top-level entries: {'OLIVES_Dataset_Labels.zip', 'OLIVES.zip'}
First 20 files:
  OLIVES_Dataset_Labels.zip
  OLIVES.zip
Extensions: {'.zip': 2}

ZIP: OLIVES_Dataset_Labels.zip
Size: 0.01 GB
Total files: 10
Top-level entries: {'OLIVES_Dataset_Labels'}
First 20 files:
  OLIVES_Dataset_Labels/
  OLIVES_Dataset_Labels/full_labels/
  OLIVES_Dataset_Labels/ml_centric_labels/
  OLIVES_Dataset_Labels/full_labels/OCT-DR.xlsx
  OLIVES_Dataset_Labels/full_labels/OCT-DME.xlsx
  OLIVES_Dataset_Labels/full_labels/Clinical_Data_Images.xlsx
  OLIVES_Dataset_Labels/full_labels/Summaries _DR_DME_Studies.docx
  OLIVES_Dataset_Labels/ml_centric_labels/Clinical_Data_Images.xlsx
  OLIVES_Dataset_Labels/full_labels/Biomarker_Clinical_Data_Images.csv
  OLIVES_Dataset_Labels/ml_centric_labels/Biomarker_Clinical_Data_Images.csv
Extensions: {'.xlsx': 4, '.docx': 1, '.csv': 2}


## Extract OLIVES labels zip

The labels zip is small (~10 MB), so we extract it directly to Drive. Required for the next cell which reads the biomarker CSV.

In [9]:
# ── PHASE 1 | OLIVES LABELS ── Extract label files from zip to Drive
# Purpose: Extract the OLIVES labels zip (only 0.01 GB) directly to
# Drive so we can read CSVs and Excel files without handling the
# large image zip yet.

import zipfile
import os

# Path to the labels zip on Drive
labels_zip = "/content/drive/MyDrive/OLIVES_Dataset_Labels.zip"

# Destination folder on Drive — will be created if it doesn't exist
extract_to = "/content/drive/MyDrive/olives_labels/"
os.makedirs(extract_to, exist_ok=True)

# Extract all label files (CSVs, Excel, docx)
with zipfile.ZipFile(labels_zip, 'r') as z:
    z.extractall(extract_to)
    print(f"Extracted {len(z.namelist())} files to:\n{extract_to}")

# Verify extraction — list all extracted files
print("\nExtracted files:")
for root, dirs, files in os.walk(extract_to):
    for f in files:
        print(f"  {os.path.join(root, f)}")

Extracted 10 files to:
/content/drive/MyDrive/olives_labels/

Extracted files:
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/full_labels/OCT-DR.xlsx
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/full_labels/OCT-DME.xlsx
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/full_labels/Clinical_Data_Images.xlsx
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/full_labels/Summaries _DR_DME_Studies.docx
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/full_labels/Biomarker_Clinical_Data_Images.csv
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/ml_centric_labels/Clinical_Data_Images.xlsx
  /content/drive/MyDrive/olives_labels/OLIVES_Dataset_Labels/ml_centric_labels/Biomarker_Clinical_Data_Images.csv


## Inspect biomarker CSV

Read the ML-centric biomarker CSV (the primary label file for the dissertation). Reports shape, columns, missing values, and class distribution. This file links each OLIVES image to its biomarker values, clinical measurements, and patient metadata.

In [11]:
# ── PHASE 1 | OLIVES LABELS ── Inspect key biomarker CSV
# Purpose: Read the ML-centric biomarker CSV which is the primary
# label file for our dissertation. This file links each image to
# its clinical biomarker values, DR grade, and patient metadata.
# This is what the multimodal fusion module will use later.

import pandas as pd

# ML-centric version is preferred over full_labels —
# it is pre-formatted for machine learning use
csv_path = (
    "/content/drive/MyDrive/olives_labels/"
    "OLIVES_Dataset_Labels/ml_centric_labels/"
    "Biomarker_Clinical_Data_Images.csv"
)

# Load the CSV
df = pd.read_csv(csv_path)

# Basic structure
print(f"Shape: {df.shape}")
print(f"  → {df.shape[0]} rows (images)")
print(f"  → {df.shape[1]} columns (features)")

# Column names — critical for understanding what labels are available
print(f"\nColumns ({len(df.columns)}):")
for col in df.columns.tolist():
    print(f"  {col}")

# Sample rows to understand data format
print(f"\nFirst 3 rows:")
print(df.head(3).to_string())

# Missing values — important for multimodal fusion
# (missing modalities are a core challenge in the dissertation)
print(f"\nMissing values per column:")
missing = df.isnull().sum()
print(missing[missing > 0].to_string()
      if missing.any()
      else "  No missing values found")

# Class distribution if a DR grade column exists
grade_cols = [c for c in df.columns
              if 'grade' in c.lower() or 'dr' in c.lower()
              or 'label' in c.lower()]
if grade_cols:
    print(f"\nDR-related columns found: {grade_cols}")
    for col in grade_cols:
        print(f"\nDistribution of '{col}':")
        print(df[col].value_counts().to_string())

Shape: (9408, 22)
  → 9408 rows (images)
  → 22 columns (features)

Columns (22):
  Path (Trial/Arm/Folder/Visit/Eye/Image Name)
  Scan (n/49)
  Atrophy / thinning of retinal layers
  Disruption of EZ
  DRIL
  IR hemorrhages
  IR HRF
  Partially attached vitreous face
  Fully attached vitreous face
  Preretinal tissue/hemorrhage
  Vitreous debris
  VMT
  DRT/ME
  Fluid (IRF)
  Fluid (SRF)
  Disruption of RPE
  PED (serous)
  SHRM
  Eye_ID
  BCVA
  CST
  Patient_ID

First 3 rows:
    Path (Trial/Arm/Folder/Visit/Eye/Image Name)  Scan (n/49)  Atrophy / thinning of retinal layers  Disruption of EZ  DRIL  IR hemorrhages  IR HRF  Partially attached vitreous face  Fully attached vitreous face  Preretinal tissue/hemorrhage  Vitreous debris  VMT  DRT/ME  Fluid (IRF)  Fluid (SRF)  Disruption of RPE  PED (serous)  SHRM  Eye_ID  BCVA  CST  Patient_ID
0  /TREX DME/GILA/0201GOD/V1/OD/TREXJ_000000.tif            1                                     0                 0     0               0       1 

## Inspect inner OLIVES.zip (nested in olive.zip)

`olive.zip` contains another zip called `OLIVES.zip`. We open the outer zip and stream the inner one without extracting it to disk. This gives us the inner structure (file count, top-level folders, extensions) so we can understand the layout before committing to extraction.

**Note**: this reads from a 34 GB file on Drive, expect 2-5 minutes.

In [12]:
# ── PHASE 1 | OLIVES IMAGES ── Inspect nested image zip
# Purpose: OLIVES.zip is nested inside olive.zip (33.92 GB).
# We inspect its contents WITHOUT extracting anything — we just
# open the outer zip, then open the inner zip as a stream, and
# list the filenames. This tells us the folder structure,
# image count, and modality breakdown (fundus vs OCT) before
# we commit to any preprocessing approach.
#
# NOTE: This cell may take a few minutes because it reads the
# zip index from a 33 GB file on Drive. Be patient.

import zipfile
import os
from collections import Counter

# Outer zip — the wrapper containing OLIVES.zip and labels zip
outer_zip_path = "/content/drive/MyDrive/olive.zip"

print("Opening outer zip (olive.zip)...")
print("This may take a few minutes — reading from Drive...\n")

with zipfile.ZipFile(outer_zip_path, 'r') as outer:

    # Open the inner OLIVES.zip as a file stream
    # without extracting it to disk first
    print("Opening inner OLIVES.zip as stream...")
    with outer.open("OLIVES.zip") as inner_file:
        with zipfile.ZipFile(inner_file, 'r') as inner:

            # Get all filenames inside the inner zip
            files = inner.namelist()

            # Basic counts
            print(f"Total files in OLIVES.zip: {len(files)}")

            # Top-level folder structure
            top_level = sorted(set(
                f.split('/')[0] for f in files
            ))
            print(f"\nTop-level folders/entries:")
            for entry in top_level:
                print(f"  {entry}")

            # First 20 filenames to understand layout
            print(f"\nFirst 20 filenames:")
            for f in files[:20]:
                print(f"  {f}")

            # File extension breakdown
            # Critical — tells us how many fundus vs OCT images
            exts = Counter(
                os.path.splitext(f)[1].lower()
                for f in files
                if os.path.splitext(f)[1]
            )
            print(f"\nFile extensions (modality breakdown):")
            for ext, count in exts.most_common():
                print(f"  {ext}: {count} files")

            # Folder-level breakdown
            # Helps identify fundus/ vs OCT/ subfolders
            folders = Counter(
                '/'.join(f.split('/')[:2])
                for f in files
                if '/' in f
            )
            print(f"\nTop folder breakdown (first 15):")
            for folder, count in folders.most_common(15):
                print(f"  {folder}: {count} files")

Opening outer zip (olive.zip)...
This may take a few minutes — reading from Drive...

Opening inner OLIVES.zip as stream...
Total files in OLIVES.zip: 3

Top-level folders/entries:
  OLIVES

First 20 filenames:
  OLIVES/
  OLIVES/TREX_DME.zip
  OLIVES/Prime_FULL.zip

File extensions (modality breakdown):
  .zip: 2 files

Top folder breakdown (first 15):
  OLIVES/: 1 files
  OLIVES/TREX_DME.zip: 1 files
  OLIVES/Prime_FULL.zip: 1 files


## Inspect trial-level zips (TREX_DME.zip, Prime_FULL.zip)

Inside `OLIVES.zip` there are two more zips, one per clinical trial. We inspect both progressively to identify fundus vs OCT files and understand the per-trial folder conventions.

**Note**: this reads from a 34 GB file on Drive, expect 5-10 minutes total.

In [13]:
# ── PHASE 1 | OLIVES IMAGES ── Inspect doubly-nested zips
# Purpose: OLIVES contains two more zips inside: TREX_DME.zip and
# Prime_FULL.zip. We need to inspect both to find fundus images
# and understand the complete folder structure before extraction.
# WARNING: This reads from a 33GB file on Drive — may take 5-10 min.

import zipfile
import os
from collections import Counter

outer_zip_path = "/content/drive/MyDrive/olive.zip"

print("Opening olive.zip → OLIVES.zip → inner zips...")
print("This may take several minutes...\n")

with zipfile.ZipFile(outer_zip_path, 'r') as outer:
    with outer.open("OLIVES.zip") as olives_file:
        with zipfile.ZipFile(olives_file, 'r') as olives_zip:

            # Inspect TREX_DME.zip
            print("="*60)
            print("INSPECTING: TREX_DME.zip")
            print("="*60)
            with olives_zip.open("OLIVES/TREX_DME.zip") as trex_file:
                with zipfile.ZipFile(trex_file, 'r') as trex:
                    files = trex.namelist()
                    print(f"Total files: {len(files)}")

                    # Top level structure
                    top = sorted(set(
                        f.split('/')[0] for f in files
                    ))
                    print(f"Top-level folders: {top}")

                    # Extension breakdown
                    exts = Counter(
                        os.path.splitext(f)[1].lower()
                        for f in files
                        if os.path.splitext(f)[1]
                    )
                    print(f"Extensions: {dict(exts)}")

                    # First 20 files
                    print(f"First 20 files:")
                    for f in files[:20]:
                        print(f"  {f}")

                    # Check for fundus specifically
                    fundus = [f for f in files
                              if 'fundus' in f.lower()
                              or 'color' in f.lower()
                              or 'cfp' in f.lower()]
                    print(f"\nFundus-related files found: "
                          f"{len(fundus)}")
                    if fundus:
                        print("Sample fundus paths:")
                        for f in fundus[:5]:
                            print(f"  {f}")

            # Inspect Prime_FULL.zip
            print("\n" + "="*60)
            print("INSPECTING: Prime_FULL.zip")
            print("="*60)
            with olives_zip.open(
                "OLIVES/Prime_FULL.zip"
            ) as prime_file:
                with zipfile.ZipFile(prime_file, 'r') as prime:
                    files = prime.namelist()
                    print(f"Total files: {len(files)}")

                    # Top level structure
                    top = sorted(set(
                        f.split('/')[0] for f in files
                    ))
                    print(f"Top-level folders: {top}")

                    # Extension breakdown
                    exts = Counter(
                        os.path.splitext(f)[1].lower()
                        for f in files
                        if os.path.splitext(f)[1]
                    )
                    print(f"Extensions: {dict(exts)}")

                    # First 20 files
                    print(f"First 20 files:")
                    for f in files[:20]:
                        print(f"  {f}")

                    # Check for fundus specifically
                    fundus = [f for f in files
                              if 'fundus' in f.lower()
                              or 'color' in f.lower()
                              or 'cfp' in f.lower()]
                    print(f"\nFundus-related files found: "
                          f"{len(fundus)}")
                    if fundus:
                        print("Sample fundus paths:")
                        for f in fundus[:5]:
                            print(f"  {f}")

Opening olive.zip → OLIVES.zip → inner zips...
This may take several minutes...

INSPECTING: TREX_DME.zip
Total files: 98897
Top-level folders: ['TREX DME']
Extensions: {'.npy': 1045, '.tif': 90706, '.jpg': 4300}
First 20 files:
  TREX DME/
  TREX DME/GILA/
  TREX DME/GILA/0234GOD/
  TREX DME/GILA/0234GOD/sequence_12.npy
  TREX DME/GILA/0234GOD/V1/
  TREX DME/GILA/0234GOD/V1/OD/
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000027.tif
  TREX DME/GILA/0234GOD/V1/OD/fundus_OD_V1.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000000.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000001.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000002.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000003.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000004.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000005.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000006.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000007.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000008.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000009.tif
  TREX DME/GILA/0234GOD/V1/OD/TREXW_000010.tif
  TREX 

---

# Summary

## What was produced

- **EyePACS HuggingFace dataset** persisted to `MyDrive/` (arrow files) — 35,108 fundus images with 5-class DR severity labels, ready for preprocessing
- **OLIVES labels** extracted to `MyDrive/olives_labels/` — biomarker CSV (9,408 rows × 22 cols), clinical data Excel files, and full-labels variants
- **OLIVES structural map** documented (in cell outputs) — confirmed:
  - `olive.zip` → `OLIVES.zip` → `{TREX_DME.zip, Prime_FULL.zip}` (three-level nesting)
  - TREX DME: 98,897 files; 1,849 fundus (`fundus_OD/OS_V{visit}.tif`)
  - Prime_FULL: 68,807 files; 2,586 fundus (`fundus_OD/OS_W{week}.{tif|png}`)
  - Total ~4,435 fundus images across the two trials
- **OLIVES image data** NOT extracted — extraction is handled by `02_preprocess.ipynb` to avoid duplicate storage on Drive

## Key findings

- EyePACS dataset is straightforward (single arrow file, standard format) — no surprises
- OLIVES has triple-nested zip structure that requires careful stream-handling to inspect without extraction
- Two distinct trial conventions inside OLIVES (TREX uses `V{visit}` folders, Prime uses `W{week}` folders) — must be handled separately in preprocessing
- Prime_FULL stores some fundus images in both `.tif` and `.png` formats — this becomes Bug 3 in the preprocessing pipeline

## What to do next

Run `02_preprocess.ipynb` to extract and preprocess both datasets into the normalised tensor format used by the rest of the project.
